pds_doi_core/db_util/doi_database.py

```diff
@@ -0,0 +1,422 @@
+
+        return 1
+
+    def doi_select_rows_all(self,db_name,table_name):
+        ''' Select all rows. '''
+        logger.debug(f"self.m_my_conn {self.m_my_conn}")
+
+        if self.m_my_conn is None:
+            logger.warn(f"Connection is None in database {self.get_database_name()}")
+            self.m_my_conn = self.doi_create_connection(db_file)
+        o_table_exist_flag = self.check_if_table_exist(table_name)
+        logger.debug(f"table_name,o_table_exist_flag {table_name},{o_table_exist_flag}")
+
+        query_string = 'SELECT * FROM ' + table_name + ';   '
+        logger.debug(f"doi_select_rows_all:query_string {query_string}")
+
+        cursor = self.m_my_conn.cursor()
+        cursor.execute(query_string)
+        column_names = list(map(lambda x: x[0], cursor.description))
+        records = cursor.fetchall()
+
+        logger.debug("len(records)",len(records))
+        for row_index, row in enumerate(records):
+            for col in range(0,10):
+             logger.debug("row_index,col,column_names[col],type(row[col]),row[col]",row_index,col,column_names[col],type(row[col]),row[col])
+
+        return 1
+
+    def doi_update_row(self,db_name,table_name,update_list,query_criterias):
+        ''' Update all rows and fields (specified in update_list matching query_criterias.'''
+
+        logger.debug(f"self.m_my_conn {self.m_my_conn}")
+
+        if self.m_my_conn is None:
+            logger.warn(f"Connection is None in database {self.get_database_name()}")
+            self.m_my_conn = self.doi_create_connection(db_file)
+        o_table_exist_flag = self.check_if_table_exist(table_name)
+        logger.debug(f"table_name,o_table_exist_flag {table_name},{o_table_exist_flag}")
+
+#       if o_table_exist_flag:
+#            logger.warn(f"Table {table_name} already exist")
+#            if (drop_exist_table_flag):
+#                logger.debug(f"drop_exist_table_flag is True for table {table_name}")
+#                self.doi_drop_table(db_name,table_name)
+
+        query_string = 'UPDATE ' + table_name + ' SET '
+        for ii in range(len(update_list)):
+            # Build the SET column_1 = new_value_1,
+            #               column_2 = new_value_2
+            # Only preceed the comma if not the first one.
+            if ii == 0:
+                query_string += update_list[ii]
+            else:
+                query_string += ',' + update_list[ii]
+
+        # Add any query_criterias
+        if len(query_criterias) > 0:
+            query_string += ' WHERE '
+        for ii in range(len(query_criterias)):
+            if ii == 0:
+                query_string += query_criterias[ii]
+            else:
+                query_string += ' AND ' + query_criterias[ii]
+        logger.debug(f"query_string {query_string}")
+
+        self.m_my_conn.execute(query_string)
+
+        return 1
+
+    def doi_delete_row(self,db_name,table_name,query_criterias):
```

Comment:
we don't need that

Reply:
Function doi_delete_row() has been removed from doi_database.py file.

Reply:
thanks